javascript/javascript3/week1/exercise 3/Exercise 3.js

```diff
@@ -0,0 +1,96 @@
+const wrapper = document.querySelector(".wrapper"),
+inputPart = document.querySelector(".input-part"),
+infoTxt = inputPart.querySelector(".info-txt"),
+inputField = inputPart.querySelector("input"),
+locationBtn = inputPart.querySelector("button"),
+weatherPart = wrapper.querySelector(".weather-part"),
+wIcon = weatherPart.querySelector("img"),
+arrowBack = wrapper.querySelector("header i");
+
+let api;
+
+inputField.addEventListener("keyup", e =>{
+    // if user pressed enter btn and input value is not empty
+    if(e.key == "Enter" && inputField.value != ""){
+        requestApi(inputField.value);
+    }
+});
+
+locationBtn.addEventListener("click", () =>{
+    if(navigator.geolocation){ // if browser support geolocation api
+        navigator.geolocation.getCurrentPosition(onSuccess, onError);
+    }else{
+        alert("Your browser not support geolocation api");
+    }
+});
+
+function requestApi(city){
+    api = `https://api.openweathermap.org/data/2.5/weather?q=${city}&units=metric&appid=9a027e91e4ffe924a9d3cde1bf9a4fcb`;
+    fetchData();
+}
+
+function onSuccess(position){
+    const {latitude, longitude} = position.coords; // getting lat and lon of the user device from coords obj
+    api = `https://api.openweathermap.org/data/2.5/weather?lat=${latitude}&lon=${longitude}&units=metric&appid=9a027e91e4ffe924a9d3cde1bf9a4fcb`;
+    fetchData();
+}
+
+function onError(error){
+    // if any error occur while getting user location then we'll show it in infoText
+    infoTxt.innerText = error.message;
+    infoTxt.classList.add("error");
+}
+
+function fetchData(){

```

Comment:
You could also use `async` `await` similar to what you did in exercise 2. In general, always try to be consistent with naming, and code style.